npm linting github action throwing error

The npm linter throws an error when it runs in the github action under:
https://github.com/codeforpdx/opentransit-metrics/blob/master/.github/workflows/test.yml
I disabled the workflow for now but it would be good to figure out the issue and fix it.

See this action as an example of the error:
https://github.com/codeforpdx/opentransit-metrics/runs/7928966223?check_suite_focus=true

It can be recreated locally:
```
docker run opentransit/react-dev:latest npm run lint:check

> opentransit-metrics@0.5.0 lint:check /app/frontend
> eslint src --ext .jsx,.js


/app/frontend/src/actions/index.js
  347:1   error    Delete `··`                               prettier/prettier
  348:1   error    Delete `··`                               prettier/prettier
  349:3   error    Delete `··`                               prettier/prettier
  351:1   error    Delete `··`                               prettier/prettier
  352:35  error    Insert `{`                                prettier/prettier
  353:7   error    Delete `{`                                prettier/prettier
  355:30  error    Insert `,`                                prettier/prettier
  356:1   error    Replace `······}·` with `····})`          prettier/prettier
  357:5   error    Delete `)`                                prettier/prettier
  360:7   error    Unexpected var, use let or const instead  no-var
  360:58  error    Insert `;`                                prettier/prettier
  361:7   error    Unexpected var, use let or const instead  no-var
  361:42  error    Insert `;`                                prettier/prettier
  362:19  error    Insert `;`                                prettier/prettier
  363:36  error    Insert `;`                                prettier/prettier
  364:16  error    Insert `;`                                prettier/prettier
  365:17  error    Insert `;`                                prettier/prettier
  366:61  error    Insert `;`                                prettier/prettier
  368:1   error    Replace `····⏎··};` with `}`              prettier/prettier
  375:5   warning  Unexpected console statement              no-console
  601:5   warning  Unexpected console statement              no-console

/app/frontend/src/components/RouteSummary.jsx
    8:1  error    `@material-ui/core/Button` import should occur before import of `./SummaryRow`  import/order
   50:1  error    Delete `⏎`                                                                      prettier/prettier
   55:5  warning  Unexpected console statement                                                    no-console
  217:7  error    Insert `··`                                                                     prettier/prettier
  218:7  error    Insert `··`                                                                     prettier/prettier
  219:1  error    Replace `······` with `········`                                                prettier/prettier
  220:1  error    Insert `··`                                                                     prettier/prettier
  222:1  error    Insert `··`                                                                     prettier/prettier

✖ 29 problems (26 errors, 3 warnings)
  26 errors and 0 warnings potentially fixable with the `--fix` option.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! opentransit-metrics@0.5.0 lint:check: `eslint src --ext .jsx,.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the opentransit-metrics@0.5.0 lint:check script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-08-26T03_50_00_624Z-debug.log
```

--- a/.githooks/pre-commit
+++ b/.githooks/pre-commit
@@ -1,0 +1,29 @@
+#!/bin/bash
+
+cd "$(git rev-parse --show-toplevel)"
+
+# Only run eslint check if staged changes include relevant files
+STAGED_FILES=($(git diff --cached --name-only --diff-filter=ACM | grep ".jsx\{0,1\}$"))
+
+if [ -z "$STAGED_FILES" ]; then
+  exit 0
+fi
+
+echo "Staged changes contain js files. Running pre-commit eslint check."
+
+# Run check in docker container since dependencies may not be installed locally
+DEV_DIR=$(cd `dirname $0` && pwd)
+ROOT_DIR=`dirname $DEV_DIR`
+
+docker run -v "$ROOT_DIR/frontend/src:/app/frontend/src" opentransit-metrics-react-dev:latest node_modules/.bin/eslint src --ext .jsx,.js
+
+ESLINTCHECK_EXIT="$?"
+
+if [[ "${ESLINTCHECK_EXIT}" == 0 ]]; then
+  printf "\n\033[42mCOMMIT SUCCEEDED\033[0m\n"
+else
+  printf "\n\033[41mCOMMIT FAILED:\033[0m Fix eslint errors manually or by running dev/docker-lint.sh and try again\n"
+  exit 1
+fi
+
+exit $?

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -13,3 +13,5 @@ jobs:
       run: docker run opentransit/flask:latest python -m unittest discover -s tests
     - name: Build Docker react-dev image
       run: docker build --target react-dev -t opentransit/react-dev:latest .
+    - name: Run lint:check
+      run: docker run opentransit/react-dev:latest npm run lint:check

--- a/README.md
+++ b/README.md
@@ -69,6 +69,12 @@ Before committing, run `dev/docker-lint.sh` (Mac/Linux) or `dev\docker-lint.bat`
 
 GitHub automatically runs tests for each push to check for eslint errors. If eslint reports any style errors, pull requests will show a failing check.
 
+You can configure git to look for any git hooks found in the `.githooks` directory for this project by running the following:
+```sh
+git config --local core.hooksPath .githooks
+```
+Existing githooks include a pre-commit hook that runs eslint automatically if changes include any js files.
+
 ### Deploying to Heroku
 
 When you make a Pull Request, we would suggest you deploy your branch to Heroku so that other

--- a/frontend/src/actions/index.js
+++ b/frontend/src/actions/index.js
@@ -318,7 +318,7 @@ export function fetchDownload(params) {
   const dates = computeDates(params.firstDateRange);
   // console.log('in fetchDownload found dates', dates);
   // console.log('getting dates[0]', dates[0]);
-  const routeId = params.routeId ;
+  const routeId = params.routeId;
 
   const variables = {
     agencyId: Agencies[0].id,
@@ -334,7 +334,7 @@ export function fetchDownload(params) {
 
   const variablesJson = JSON.stringify(variables);
 
-  console.log('routeId', routeId);  
+  console.log('routeId', routeId);
 
   let downloadFilename;
 
@@ -345,29 +345,33 @@ export function fetchDownload(params) {
     downloadFilename = `arrivals_${routeId}_${dates[0]}_${dates[1]}.csv`;
   }
 
+  console.log(
+    'inside fetchDownload action routeId',
+    params.routeId,
+    'directionId',
+    params.directionId,
+    'dates',
+    dates,
+  );
 
-    console.log('inside fetchDownload action routeId', params.routeId
-    , 'directionId', params.directionId, 'dates', dates);
-
-    axios.get(
-      '/api/arrival_download', 
-      {responseType: 'blob',
+  axios
+    .get('/api/arrival_download', {
+      responseType: 'blob',
       params: { variables: variablesJson },
       baseURL: MetricsBaseURL,
-    } 
-    ).then((response) => {
+    })
+    .then(response => {
       // copied from https://stackoverflow.com/questions/65212805/how-can-i-open-csv-file-received-from-axios-get-response-in-new-window
       // there might be a better way to download the file
-      var url = window.URL.createObjectURL(response.data)
-      var a = document.createElement('a')
-      a.href = url
-      a.download = downloadFilename
-      a.click()
-      a.remove()
-      setTimeout(() => window.URL.revokeObjectURL(url), 100)
+      const url = window.URL.createObjectURL(response.data);
+      const a = document.createElement('a');
+      a.href = url;
+      a.download = downloadFilename;
+      a.click();
+      a.remove();
+      setTimeout(() => window.URL.revokeObjectURL(url), 100);
     });
-    
-  };
+}
 
 export function fetchRouteMetrics(params) {
   const dates = computeDates(params.firstDateRange);

--- a/frontend/src/components/RouteSummary.jsx
+++ b/frontend/src/components/RouteSummary.jsx
@@ -1,11 +1,11 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import { Table, TableBody, TableHead } from '@material-ui/core';
+import Button from '@material-ui/core/Button';
 import SummaryRow from './SummaryRow';
 import SummaryHeaderRow from './SummaryHeaderRow';
 import { metersToMiles } from '../helpers/routeCalculations';
 import { renderDateRange } from '../helpers/dateTime';
-import Button from '@material-ui/core/Button';
 import { fetchDownload } from '../actions/index';
 
 /*
@@ -46,7 +46,6 @@ function RouteSummary(props) {
     intervalMetrics = routeIntervalMetrics;
     intervalMetrics2 = routeIntervalMetrics2;
   }
-
 
   function handleDownload() {
     // setDateRangeParams(localDateRangeParams);
@@ -214,12 +213,12 @@ function RouteSummary(props) {
         </TableBody>
       </Table>
       <Button
-      onClick={handleDownload}
-      color="primary"
-      variant="contained"
-      // disabled={}
+        onClick={handleDownload}
+        color="primary"
+        variant="contained"
+        // disabled={}
       >
-      Download Arrival Data
+        Download Arrival Data
       </Button>
     </>
   );

